jetty-util/src/main/java/org/eclipse/jetty/util/thread/ReservedThreadExecutor.java

```diff
@@ -266,139 +254,136 @@ public String toString()
         return String.format("%s@%x{s=%d/%d,p=%d}",
             getClass().getSimpleName(),
             hashCode(),
-            _size.get(),
+            _count.getLo(),
             _capacity,
-            _pending.get());
+            _count.getHi());
     }
 
-    private class ReservedThread implements Runnable
+    private enum State
     {
-        private final SynchronousQueue<Runnable> _task = new SynchronousQueue<>();
-        private boolean _starting = true;
-
-        public boolean offer(Runnable task)
-        {
-            if (LOG.isDebugEnabled())
-                LOG.debug("{} offer {}", this, task);
-
-            try
-            {
-                _task.put(task);
-                return true;
-            }
-            catch (Throwable e)
-            {
-                LOG.ignore(e);
-                _size.getAndIncrement();
-                _stack.offerFirst(this);
-                return false;
-            }
-        }
+        PENDING,
+        POLLING,
+        RUNNING,
+        IDLED_OUT,
+        STOPPED
+    }
 
-        public void stop()
-        {
-            offer(STOP);
-        }
+    private class ReservedThread implements Runnable
+    {
+        private final AtomicReference<State> _state = new AtomicReference<>(State.PENDING);
```

Comment:
`_state` could be a simple `volatile` as only `toString()` makes it visible to other threads.

Reply:
probably... but essentially AtomicReference is just a simple volatile with some conveniance methods :)
I'll have a play and see if it reads better.